hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveViewCatalog.java

```diff
@@ -125,6 +125,34 @@ public void testHiveViewAndIcebergViewWithSameName() throws TException, IOExcept
         .hasMessageStartingWith("Not an iceberg view: hive.hivedb.test_hive_view");
   }
 
+  @Test
+  public void testHiveViewExists() {
+    String dbName = "hivedb";
+    Namespace ns = Namespace.of(dbName);
+    String viewName = "test_hive_view_exists";
+    TableIdentifier identifier = TableIdentifier.of(ns, viewName);
+    TableIdentifier invalidIdentifier = TableIdentifier.of(dbName, "invalid", viewName);
+    if (requiresNamespaceCreate()) {
+      catalog.createNamespace(identifier.namespace());
+    }
+
+    assertThat(catalog.viewExists(invalidIdentifier))
+        .as("Should return false on invalid view identifier")
+        .isFalse();
+    assertThat(catalog.viewExists(identifier)).as("View should not exist before create").isFalse();
```

Comment:
I feel like this unnecessarily duplicates the tests that are being done in https://github.com/apache/iceberg/blob/main/core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java#L77

The only difference here is that you're also checking view existence with an invalid identifier

Reply:
I believe what would be good to test here is if you check view existence when the identifier
* is actually an iceberg table
* is a non-iceberg view